TypeCobol/Compiler/CodeElements/Expressions/SymbolName.cs

```diff
@@ -166,6 +177,33 @@ public override bool AcceptASTVisitor(IASTVisitor astVisitor)
             return base.AcceptASTVisitor(astVisitor) && astVisitor.Visit(this);
         }
 
+        public override bool Equals(object obj)
+        {
+            return this.Equals(obj as SymbolReference);
+        }
+
+        public bool Equals(SymbolReference symbolReferenceCompare)
+        {
+            if (Object.ReferenceEquals(this, symbolReferenceCompare)) return true;
+            if (Object.ReferenceEquals(null, symbolReferenceCompare)) return false;
+
+            return Type == symbolReferenceCompare.Type &&
+                   NameLiteral == symbolReferenceCompare.NameLiteral &&
+                   Role == symbolReferenceCompare.Role;
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = 13;
+                hashCode = (hashCode * 397) ^ Type.GetHashCode();
+                hashCode = (hashCode * 397) ^ NameLiteral.GetHashCode();
+                hashCode = (hashCode * 397) ^ Role.GetHashCode();
+
+                return hashCode;
+            }
+        }
```

Comment:
SymbolReference inherits from SymbolInformation so it seems odd to use the Role here and not in the base class. Could you check whether it works or not without overriding anything in SymbolReference, just using the SymbolInformation comparison mechanism ?